feeder/feeds/collectors/hazard_monitoring.py

```diff
@@ -0,0 +1,171 @@
+    And few features are annotated into it
+    """
+
+    IDMC_HAZARD_TRANSLATOR = 'https://raw.githubusercontent.com/idmc-labs/idmc-monitoring-platform/master/hazard_monitoring/IDMC_Hazard_Types_Translator.csv'
+    SOURCE = 'GDACS'
+    SOURCE_TYPE = 'Disaster'
+
+    # geohash precision
+    GEOHASH_PRECISION = 9
+
+    # gdacs keys
+    EVENT_TYPE = 'eventtype'
+    GDACS_EVENT_DESCRIPTION = 'description'
+    GDACS_COUNTRY = 'country'
+    GDACS_FROM_DATE = 'fromdate'
+    # idmc keys
+    IDMC_EVENT_NAME = 'idmc_event_name'
+    IDMC_GDACS_EVENT_TYPE = 'GDACS_eventtype'
+    IDMC_HAZARD_ID = 'IDMC_id'
+    OUTPUT_IDMC_HAZARD_ID = 'hazard_type_id'
+    IDMC_HAZARD_TYPE = 'IDMC_type'
+    OUTPUT_IDMC_HAZARD_TYPE = 'hazard_type'
+    # new keys
+    DISPLACEMENT_MENTIONED = 'displacement_mentioned'
+    LOCATION_ID = 'location_id'
+    # to transform dates to ISO
+    DATE_FEATURES = ('fromdate', 'pubDate', 'todate')
+    # key transformation
+    KEY_TRANSFORM = {
+        'fromdate': 'start_date',
+        'glide': 'glide_number',
+        'todate': 'end_date',
+        'pubDate': 'pub_date',
+        'eventid': 'event_id',
+        'eventname': 'event_name',
+        'eventname': 'event_name',
+        'eventtype': 'hazard_code_source',
+        'alertlevel': 'alert_score',
+        'population': 'affected',
+        'country': 'countries_affected',
+        'link': 'source_url',
+    }
+
+    def fetch_hazard_types(self):
+        r = requests.get(self.IDMC_HAZARD_TRANSLATOR)
+        reader = csv.reader([each.decode() for each in r.iter_lines()])
+        headers = next(reader)
+        hazard_types = []
+        for line in reader:
+            hazard_types.append(dict(zip(headers, line)))
+        return hazard_types
+
+    def map_idmc_hazard(self, items: List):
+        hazard_types = self.fetch_hazard_types()
+        for item in items:
+            idmc_hazard = list(filter(lambda x: x[self.IDMC_GDACS_EVENT_TYPE] == item[self.EVENT_TYPE], hazard_types))
+            item[self.OUTPUT_IDMC_HAZARD_ID] = None
+            item[self.OUTPUT_IDMC_HAZARD_TYPE] = None
+            if idmc_hazard:
+                item[self.OUTPUT_IDMC_HAZARD_ID] = idmc_hazard[0][self.IDMC_HAZARD_ID]
+                item[self.OUTPUT_IDMC_HAZARD_TYPE] = idmc_hazard[0][self.IDMC_HAZARD_TYPE]
+
+    def annotate_is_displacement_mentioned(self, item: dict) -> bool:
+        return {self.DISPLACEMENT_MENTIONED: bool(re.search('displace|destro|idp', item[self.GDACS_EVENT_DESCRIPTION]))}
+
+    def get_closest_neighbor(self, item: dict) -> dict:
+        """
+        NeighborFinder using WD (I dont know whats WD)
+        attributes annotated : OBJECTID, ISO3, ISO2, Short_name,lat, log, name, region, iso
+
+        gdacs_country = countries affected (one or more)
+        short_name (above) pins down to a single country
+        
+        MAYBE: mainly to point out which country the event lies in
+        """
+        # todo
+        {
```

Comment:
Did you mean to return this? or it's a shorthand?

Reply:
Forgot the return. :laughing:  Still under todo though.